rosbag2_transport/src/rosbag2_transport/recorder.cpp

```diff
@@ -106,13 +106,22 @@ void Recorder::subscribe_topics(
 
 void Recorder::subscribe_topic(const rosbag2::TopicMetadata & topic)
 {
+  if(writer_) {
```

Comment:
this check shouldn't be necessary because `writer_` stays a initialized shared pointer.